vunit/test_runner.py

```diff
@@ -210,6 +210,9 @@ def read_output():
             results = test_suite.run(output_path=output_path,
                                      read_output=read_output)
         except KeyboardInterrupt:
+            for name in test_suite.test_cases:
+                results[name] = SKIPPED
+                self._add_results(test_suite, results, start_time, num_tests, output_file_name)
```

Comment:
_add_result should be called once for the test suite and not once for every test case.